scripts/rosdistro_freeze_source

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+from rosdistro.index import Index
+
+import argparse
+import os.path
+import subprocess
+import sys
+import threading
+import time
+import yaml
+
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+
+from rosdistro import get_distribution
+from rosdistro.writer import yaml_from_distribution_file
+
+CONCURRENT_OPERATIONS=16
```

Comment:
Unsure if it's worth parameterizing this. 16 seems like a reasonable balance between parallelism and not hammering your git server (or Github).